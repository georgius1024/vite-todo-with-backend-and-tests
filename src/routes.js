import Todos from "./views/Todos.vue";
import Todo from "./views/Todo.vue";
import TodoNew from "./views/TodoNew.vue";

export default [
  { path: "/new", component: TodoNew },
  { path: "/:id", component: Todo },
  { path: "/", component: Todos },
];
