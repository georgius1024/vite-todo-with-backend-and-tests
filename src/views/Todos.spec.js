import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import component from "./Todos.vue";
import { createTestingPinia } from '@pinia/testing'
import { createRouter, createWebHistory } from "vue-router";
import { flushPromises } from "@vue/test-utils";
import routes from "../routes";

const router = createRouter({
  history: createWebHistory(),
  routes,
});

vi.mock("../services/todos", () => ({
  default: function Todos() {
      this.index = vi.fn().mockResolvedValue({data: [{ id: 1 }, { id: 2 }, { id: 3 }]});
    },
}));


describe("Todos page", () => {
  beforeEach(async () => {
    router.push("/");
    await router.isReady();
  });

  const global = {
    plugins: [router, createTestingPinia()]
  }
  it("mounts without error", async () => {
    const wrapper = mount(component, { global });
    await flushPromises()
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("button click navigates", async () => {
    const push = vi.spyOn(router, 'push');
    const wrapper = mount(component, { global });
    await flushPromises()
    await wrapper.find("a").trigger("click");
    expect(push).toBeCalledWith('/new');
  });
});
