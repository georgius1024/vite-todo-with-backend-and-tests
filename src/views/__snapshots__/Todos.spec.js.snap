// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todos page > mounts without error 1`] = `
"<div>
  <h1>Todos</h1>
  <div data-v-e895cfd1=\\"\\">
    <div data-v-e895cfd1=\\"\\" class=\\"list-group\\"></div>
    <ul data-v-e895cfd1=\\"\\" class=\\"pagination mt-4\\"></ul>
  </div><a href=\\"/new\\" class=\\"btn btn-secondary\\" role=\\"button\\">New</a>
</div>"
`;
